### PR TITLE
error handling by custom router

### PR DESCRIPTION
**Pull request by yarinTag**

### CODE REVIEW

File: src/modules/departments/department.service.ts [hunk 1]
```diff
@@ -31,7 +32,7 @@ export class DepartmentService {
     const validationResult = await validationEntity(Department, department);
 
     if (validationResult.sucsses === false) {
-      throw new Error(
+      throw new BadRequestError(
```
Comment (yariv245) [inline src/modules/departments/department.service.ts]:
```suggestion
      throw new NotFoundError (
```

File: src/modules/departments/department.service.ts [hunk 2]
```diff
@@ -48,14 +49,14 @@ export class DepartmentService {
     });
 
     if (!department) {
-      throw new Error(`Departments with Id: ${req.id} not found`);
+      throw new BadRequestError(`Departments with Id: ${req.id} not found`);
```
Comment (yariv245) [inline src/modules/departments/department.service.ts]:
```suggestion
      throw new NotFoundError(`Departments with Id: ${req.id} not found`);
```

File: src/modules/departments/department.service.ts [hunk 3]
```diff
@@ -75,11 +76,11 @@ export class DepartmentService {
     });
 
     if (!department) {
-      throw new Error(`Departments with Id: ${req.id} not found`);
+      throw new UnprocessableEntityError(`Departments with Id: ${req.id} not found`);
```
Comment (yariv245) [inline src/modules/departments/department.service.ts]:
```suggestion
      throw new NotFoundError (`Departments with Id: ${req.id} not found`);
```

File: src/modules/departments/department.service.ts [hunk 4]
```diff
@@ -48,14 +49,14 @@ export class DepartmentService {
...
-      throw new Error(`Departments with Id: ${req.id} not found`);
+      throw new BadRequestError(`Departments with Id: ${req.id} not found`);
     }
 
     const entity = plainToInstance(Department, { ...department, ...req });
     const validationResult = await validationEntity(Department, entity);
 
     if (validationResult.sucsses === false) {
-      throw new Error(
+      throw new BadRequestError(
```
Comment (yariv245) [inline src/modules/departments/department.service.ts]:
```suggestion
      throw new UnprocessableEntityError(
```

File: src/modules/users/user.controller.ts [hunk 5]
```diff
@@ -9,13 +13,10 @@ class UserController {
...
     res: Response
   ): Promise<Response> => {
-    try {
-      const newUser = await this.userService.createUser(req.body);
-      return res.status(201).json(newUser);
-    } catch (error) {
-      console.error(error);
-      return res.status(500).json({ error: 'Error creating user' });
-    }
+    const newUser = await this.userService.createUser(req.body);
```
Comment (yariv245) [inline src/modules/users/user.controller.ts:16]: missing here validateEntity

**Review by yariv245 — APPROVED** [review]

File: src/routes/customRouter.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import { Router, RequestHandler, ErrorRequestHandler } from 'express';
+import { asyncWrapper } from '../middlewares/error/asyncWrapper';
+
+class AsyncRouter {
```
Comment (yariv245) [inline src/routes/customRouter.ts]: change the file name to the class name

**Review by yariv245 — APPROVED** [review]

File: src/modules/users/user.controller.ts (hunk 5 above)
Comment (yarinTag) [reply src/modules/users/user.controller.ts:16]: it's inside the user service create function

**Pull request merged**